src/zap/zapinfo.cpp

```diff
@@ -2102,30 +2104,28 @@ void ZapInfo::GetProfilingHandle(BOOL                      *pbHookFunction,
     *pbIndirectedHandles = TRUE;
 }
 
-DWORD FilterMethodAttribsForIsSupported(DWORD attribs, CORINFO_METHOD_HANDLE ftn, ICorDynamicInfo* pJitInfo)
+//
+// This strips the CORINFO_FLG_JIT_INTRINSIC flag from some of the hardware intrinsic methods.
+//
+DWORD FilterHardwareIntrinsicMethodAttribs(DWORD attribs, CORINFO_METHOD_HANDLE ftn, ICorDynamicInfo* pJitInfo)
 {
     if (attribs & CORINFO_FLG_JIT_INTRINSIC)
     {
-        // Do not report the get_IsSupported method as an intrinsic. This will turn the call into a regular
-        // call. We also make sure none of the hardware intrinsic method bodies get pregenerated in crossgen
-        // (see ZapInfo::CompileMethod) but get JITted instead. The JITted method will have the correct
-        // answer for the CPU the code is running on.
-
+        // Figure out which intrinsic we are dealing with.
         const char* namespaceName;
         const char* className;
         const char* enclosingClassName;
         const char* methodName = pJitInfo->getMethodNameFromMetadata(ftn, &className, &namespaceName, &enclosingClassName);
 
-        // If it's not the IsSupported method, we're done
-        if (strcmp(methodName, "get_IsSupported") != 0)
-            return attribs;
-
-        bool isX86intrinsic = strcmp(namespaceName, "System.Runtime.Intrinsics.X86") == 0;
+        // Is this the get_IsSupported method that checks whether intrinsic is supported?
+        bool fIsGetIsSupportedMethod = strcmp(methodName, "get_IsSupported") == 0;
 
 #if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
-        // If it's IsSupported on Sse/Sse2, we can expand unconditionally since this is reliably
+        bool fIsX86intrinsic = strcmp(namespaceName, "System.Runtime.Intrinsics.X86") == 0;
+
+        // If it's anything related to Sse/Sse2, we can expand unconditionally since this is reliably
```

Comment:
nit: It would probably be better to say: "since this is a baseline requirement for CoreCLR".